### PR TITLE
Implement import conversion logic for GZ and Zip plugins

### PR DESCRIPTION
**Pull request by hammady**
- Implement import conversion logic for GZ and Zip plugins
- Add tests for the new code
- Add Dockerfile
- Add instructions on how to build, test and publish

### CODE REVIEW

File: rayyan-formats-plugins.gemspec [hunk 1]
```diff
@@ -35,7 +35,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'coderay', '~> 1.1'
   spec.add_development_dependency 'coveralls', '~> 0.8'
 
-  spec.add_dependency 'rayyan-formats-core', "~> 0.2.1"
+  spec.add_dependency 'rayyan-formats-core', "~> 0.3"
```
Comment (hammady) [inline rayyan-formats-plugins.gemspec:38]: @fabubaker Note the change here to allow minor versions starting from `0.3`.

File: lib/rayyan-formats-plugins/plugins/gz.rb [hunk 2]
```diff
@@ -17,9 +17,10 @@ class GZ < RayyanFormats::Base
         gzipped = Zlib::GzipReader.new(fileContent)
         fileContent = StringIO.new(gzipped.read).string
         gzipped.close
-        plugin.do_import(fileContent, filename, &block)
+        fileContent = converter.call(fileContent, ext) if converter
```
Comment (fabubaker) [inline lib/rayyan-formats-plugins/plugins/gz.rb:20]: Isn't the convention for this repo to use snake_case?

Comment (hammady) [reply lib/rayyan-formats-plugins/plugins/gz.rb:20]: I didn't want to do wider scope changes, but yeah I can fix that.

Comment (hammady) [reply lib/rayyan-formats-plugins/plugins/gz.rb]: Fixed.

**Review by fabubaker — APPROVED** [review]

**Pull request merged**